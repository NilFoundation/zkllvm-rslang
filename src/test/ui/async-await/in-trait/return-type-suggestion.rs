// edition: 2021

#![feature(async_fn_in_trait)]
//~^ WARN the feature `async_fn_in_trait` is incomplete and may not be safe to use and/or cause compiler crashes

trait A {
    async fn e() {
        Ok(())
        //~^ ERROR mismatched types
        //~| HELP consider using a semicolon here
    }
}

fn main() {}
