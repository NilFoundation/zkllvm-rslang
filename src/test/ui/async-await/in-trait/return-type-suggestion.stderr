warning: the feature `async_fn_in_trait` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/return-type-suggestion.rs:3:12
   |
LL | #![feature(async_fn_in_trait)]
   |            ^^^^^^^^^^^^^^^^^
   |
   = note: see issue #91611 <https://github.com/rust-lang/rust/issues/91611> for more information
   = note: `#[warn(incomplete_features)]` on by default

error[E0308]: mismatched types
  --> $DIR/return-type-suggestion.rs:8:9
   |
LL |         Ok(())
   |         ^^^^^^- help: consider using a semicolon here: `;`
   |         |
   |         expected `()`, found enum `Result`
   |
   = note: expected unit type `()`
                   found enum `Result<(), _>`

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0308`.
